File: CS/Form1.cs
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UserDesigner;
using System;
using System.Drawing.Design;
using System.Windows.Forms;

namespace WinFormsApp_CustomNumericLabel
{
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private void button1_Click(object sender, EventArgs e)
        {
            ShowDesignerWithCustomControl();
        }
        void ShowDesignerWithCustomControl()
        {
            // Creates a Design Tool instance with the specified report instance.
            ReportDesignTool designTool = new ReportDesignTool(new XtraReport1());
            IDesignForm designForm = designTool.DesignForm;
            designForm.DesignMdiController.DesignPanelLoaded += DesignMdiController_DesignPanelLoaded;
            designTool.ShowDesignerDialog();
        }

        void DesignMdiController_DesignPanelLoaded(object sender, DesignerLoadedEventArgs e) {
            IToolboxService toolboxService =
                (IToolboxService)e.DesignerHost.GetService(typeof(IToolboxService));
            // Removes the XRLabel toolbox item.
            toolboxService.RemoveToolboxItem(
                GetToolBoxControl(toolboxService, "DevExpress.XtraReports.UI.XRLabel"));
            // Creates a new toolbox item for the custom control.
            ToolboxItem numericLabelItem = new ToolboxItem(typeof(NumericLabel))
            {
                // Specifies the name in the toolbox.
                DisplayName = "Numeric Label"
            };
            // Adds the new control to the toolbox.
            toolboxService.AddToolboxItem(numericLabelItem);
        }

        ToolboxItem GetToolBoxControl(IToolboxService toolboxService, string name) {
            foreach (ToolboxItem item in toolboxService.GetToolboxItems()) {
                if (item.TypeName == name) { return item; };
            }
            return null;
        }
    }
}

File: VB/Form1.vb
#Region "usings"
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UserDesigner
Imports System
Imports System.Drawing.Design
Imports System.Windows.Forms
#End Region

Namespace WinFormsApp_CustomNumericLabel
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			ShowDesignerWithCustomControl()
		End Sub
		#Region "ShowDesignerWithCustomControl"
		Private Sub ShowDesignerWithCustomControl()
			' Creates a Design Tool instance with the specified report instance.
			Dim designTool As New ReportDesignTool(New XtraReport1())
			Dim designForm As IDesignForm = designTool.DesignForm
			AddHandler designForm.DesignMdiController.DesignPanelLoaded, AddressOf DesignMdiController_DesignPanelLoaded
			designTool.ShowDesigner()
		End Sub

		Private Sub DesignMdiController_DesignPanelLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
			Dim toolboxService As IToolboxService = DirectCast(e.DesignerHost.GetService(GetType(IToolboxService)), IToolboxService)
			' Removes the XRLabel toolbox item.
			toolboxService.RemoveToolboxItem(GetToolBoxControl(toolboxService, "DevExpress.XtraReports.UI.XRLabel"))
			' Creates a new toolbox item for the custom control.
			Dim numericLabelItem As New ToolboxItem(GetType(NumericLabel)) With {.DisplayName = "Numeric Label"}
			' Adds the new control to the toolbox.
			toolboxService.AddToolboxItem(numericLabelItem)
		End Sub

		Private Function GetToolBoxControl(ByVal toolboxService As IToolboxService, ByVal name As String) As ToolboxItem
			For Each item As ToolboxItem In toolboxService.GetToolboxItems()
				If item.TypeName = name Then
					Return item
				End If
			Next item
			Return Nothing
		End Function
		#End Region
	End Class
End Namespace
